lib/tasks/set_missions_to_regulated.rake

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+# This namespace contains all tasks related to the mission model
+namespace :mission do # rubocop:disable Metrics/BlockLength
+  desc "set futur mission with the :name 'permanence clac' to :regulated"
+  task set_missions_to_regulated: :environment do
+    missions = Mission.where('start_date > ? and genre = ? and name = ?',
+                             DateTime.current + 1.day,
+                             0,
+                             'permanence clac')
+    missions.each do |mission|
+      unless match_all_enrollment_a_time_slot?(mission)
+        puts("the mission at #{mission.start_date} has not been setted, an enrollment mismatch the mission's timeslots")
+        next
+      end
+
+      puts("the mission at #{mission.start_date} has been setted") if mission.update(genre: 1)
+    end
+  end
+
+  def match_all_enrollment_a_time_slot?(mission)
```

Comment:
Did you mean something like
`def all_enrollments_match_a_time_slot?(mission)`
?